lib/reader/src/parser.rs

```diff
@@ -158,10 +162,31 @@ impl<'a> Context<'a> {
         self.map.def_ebb(src_ebb, ebb, loc).and(Ok(ebb))
     }
 
+    fn add_alias(&mut self, src: Value, dest: Value, loc: &Location) -> Result<()> {
+        if self.aliases.insert(src, dest).is_some() {
+            err!(loc, "duplicate alias: {} -> {}", src, dest)
+        } else {
+            Ok(())
+        }
+    }
+
     // The parser creates all instructions with Ebb and Value references using the source file
     // numbering. These references need to be rewritten after parsing is complete since forward
     // references are allowed.
     fn rewrite_references(&mut self) -> Result<()> {
+        for (&source_from, &source_to) in &self.aliases {
+            let ir_to = self.map
+                .get_value(source_to)
+                .expect(&*format!("IR destination value alias not found for {}", source_to));
```

Comment:
I've considered it, but we need the source location in order to report it as a parser error. I suppose I can add those information into the alias map, but I'm not sure if that would be the right way to go.